lib/compat/wordpress-6.0/class-wp-webfonts.php

```diff
@@ -81,19 +107,45 @@ public function get_providers() {
 	public function register_font( $font ) {
 		$font = $this->validate_font( $font );
 		if ( $font ) {
-			$id                    = $this->get_font_id( $font );
-			$this->webfonts[ $id ] = $font;
+			$slug = $this->get_font_slug( $font );
+
+			if ( ! isset( $this->registered_webfonts[ $slug ] ) ) {
+				$this->registered_webfonts[ $slug ] = array();
+			}
+
+			$this->registered_webfonts[ $slug ][] = $font;
+		}
+	}
+
+	public function enqueue_font( $font_family ) {
+		$slug = $this->get_font_slug( $font_family );
+
+		if ( isset( $this->enqueued_webfonts[ $slug ] ) ) {
+			_doing_it_wrong( __FUNCTION__, sprintf( __( 'The "%s" font family is already enqueued.' ), $slug ), '6.0.0' );
+			return false;
+		}
+
+		if ( ! isset( $this->registered_webfonts[ $slug ] ) ) {
+			_doing_it_wrong( __FUNCTION__, sprintf( __( 'The "%s" font family is not registered.' ), $slug ), '6.0.0' );
+			return false;
 		}
+
+		$this->enqueued_webfonts[ $slug ] = $this->registered_webfonts[ $slug ];
+		unset( $this->registered_webfonts[ $slug ] );
 	}
 
 	/**
-	 * Get the font ID.
+	 * Get the font slug.
 	 *
 	 * @param array $font The font arguments.
 	 * @return string
 	 */
-	public function get_font_id( $font ) {
-		return sanitize_title( "{$font['font-family']}-{$font['font-weight']}-{$font['font-style']}-{$font['provider']}" );
+	public function get_font_slug( $font ) {
+		if ( is_string( $font ) ) {
+			return sanitize_title( $font );
+		}
+
+		return sanitize_title( $font['font-family'] );
 	}
```

Comment:
Hmm, does this need to be publicly exposed within the API? Seems to be an internal function that assumes it will receive an already validated array or font-family. I'd suggest making it `private` as part of the internals.

I'd also suggest refactoring for readability and reducing the risks of future changes touching only 1 of the `sanitize_title()` instead of both.

```suggestion
	private function get_font_slug( $to_convert ) {
		if ( is_array( $to_convert ) ) {
			$to_convert = $to_convert['font-family'];
		}	

		return sanitize_title( $to_convert );
	}
```

`$to_convert` is more appropriate and descriptive than `$webfont`. Why? The param/var name of`$webfont`'s is (or should be) consistently referring to the web font's configuration (which includes it's font-face properties). Here, the method will receive a webfont's configuration (as an array) or the font-family (as a string). Since it's different kinds of data being passed into the method for conversion into a slug, a more generic name is needed for readability and future contributions.